--- a/v4l/scripts/make_kconfig.pl
+++ b/v4l/scripts/make_kconfig.pl
@@ -636,6 +636,13 @@ $config{'COMPILE_TEST'} = 0;
 $intopt { "DVB_MAX_ADAPTERS" } = 48;
 disable_config('DVB_DEMUX_SECTION_LOSS_LOG');
 disable_config('DVB_SAA716X_FF');
+disable_config('MEDIA_ANALOG_TV_SUPPORT');
+disable_config('MEDIA_CAMERA_SUPPORT');
+disable_config('MEDIA_CEC_SUPPORT');
+disable_config('MEDIA_RADIO_SUPPORT');
+disable_config('CEC_CORE');
+disable_config('DVB_MMAP');
+disable_config('VIDEO_TW9910');
 
 # Check dependencies
 my %newconfig = checkdeps();
