--- a/v4l/scripts/make_kconfig.pl
+++ b/v4l/scripts/make_kconfig.pl
@@ -633,6 +633,21 @@ $config{'COMPILE_TEST'} = 0;
 disable_config('DVB_SAA716X_FF');
 disable_config('CEC_CORE');
 disable_config('MEDIA_CEC_SUPPORT');
+disable_config('MEDIA_ANALOG_TV_SUPPORT');
+disable_config('MEDIA_CAMERA_SUPPORT');
+disable_config('MEDIA_RADIO_SUPPORT');
+disable_config('MEDIA_TEST_SUPPORT');
+disable_config('V4L_PLATFORM_DRIVERS');
+# needs HDMI
+disable_config('VIDEO_ADV7511');
+disable_config('VIDEO_ADV7604');
+disable_config('VIDEO_ADV7842');
+disable_config('VIDEO_TC358743');
+disable_config('VIDEO_TDA1997X');
+disable_config('VIDEO_TW9910');
+# needs REGMAP_SCCB
+disable_config('VIDEO_OV772X');
+disable_config('VIDEO_OV9650');
 
 # Check dependencies
 my %newconfig = checkdeps();
