--- a/linux/drivers/media/Kconfig
+++ b/linux/drivers/media/Kconfig
@@ -261,3 +261,5 @@ source "drivers/media/dvb-frontends/Kcon
 endmenu
 
 endif # MEDIA_SUPPORT
+
+source "drivers/media/vtunerc/Kconfig"
--- a/linux/drivers/media/Makefile
+++ b/linux/drivers/media/Makefile
@@ -29,6 +29,6 @@ obj-$(CONFIG_CEC_CORE) += cec/
 # Finally, merge the drivers that require the core
 #
 
-obj-y += common/ platform/ pci/ usb/ mmc/ firewire/ spi/ test-drivers/
+obj-y += common/ platform/ pci/ usb/ mmc/ firewire/ spi/ test-drivers/ vtunerc/
 obj-$(CONFIG_VIDEO_DEV) += radio/
 
