--- a/Makefile
+++ b/Makefile
@@ -71,7 +71,7 @@ OBJS = args.o audio.o channels.o ci.o co
        dvbplayer.o dvbspu.o dvbsubtitle.o eit.o eitscan.o epg.o filter.o font.o i18n.o interface.o keys.o\
        lirc.o menu.o menuitems.o nit.o osdbase.o osd.o pat.o player.o plugin.o positioner.o\
        receiver.o recorder.o recording.o remote.o remux.o ringbuffer.o sdt.o sections.o shutdown.o\
-       skinclassic.o skinlcars.o skins.o skinsttng.o sourceparams.o sources.o spu.o status.o svdrp.o themes.o thread.o\
+       skinclassic.o skins.o sourceparams.o sources.o spu.o status.o svdrp.o themes.o thread.o\
        timers.o tools.o transfer.o vdr.o videodir.o
 
 DEFINES  += $(CDEFINES)
--- a/vdr.c
+++ b/vdr.c
@@ -60,8 +60,6 @@
 #include "recording.h"
 #include "shutdown.h"
 #include "skinclassic.h"
-#include "skinlcars.h"
-#include "skinsttng.h"
 #include "sourceparams.h"
 #include "sources.h"
 #include "status.h"
@@ -832,8 +830,6 @@ int main(int argc, char *argv[])
 
   // Default skins:
 
-  new cSkinLCARS;
-  new cSkinSTTNG;
   new cSkinClassic;
   Skins.SetCurrent(Setup.OSDSkin);
   cThemes::Load(Skins.Current()->Name(), Setup.OSDTheme, Skins.Current()->Theme());
