Description: Don't add same pid multiple times
Origin: http://www.vdr-portal.de/board17-developer/board97-vdr-core/p1252658-zu-viele-pids/#post1252658
Forwarded: yes
Author: Zabrimus @ vdr-portal

--- a/receiver.c
+++ b/receiver.c
@@ -37,8 +37,10 @@ void cReceiver::SetPriority(int Priority
 bool cReceiver::AddPid(int Pid)
 {
   if (Pid) {
-     if (numPids < MAXRECEIVEPIDS)
-        pids[numPids++] = Pid;
+     if (numPids < MAXRECEIVEPIDS) {
+        if (!WantsPid(Pid))
+           pids[numPids++] = Pid;
+        }
      else {
         dsyslog("too many PIDs in cReceiver (Pid = %d)", Pid);
         return false;
