--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,7 @@ VERSION = $(shell grep 'static const char \*VERSION *=' $(PLUGIN).cpp | awk '{ p
 ### The directory environment:
 
 # Use package data if installed...otherwise assume we're under the VDR source directory:
-PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
+PKGCFG = $(shell pkg-config.real --variable=$(1) $(STAGING_DIR)/usr/lib/pkgconfig/vdr.pc)
 LIBDIR = $(DESTDIR)$(call PKGCFG,libdir)
 LOCDIR = $(DESTDIR)$(call PKGCFG,locdir)
 #
@@ -49,13 +49,13 @@ SOFILE = libvdr-$(PLUGIN).so
 
 VDRMANAGER_USE_SSL := 1
 ifneq ($(VDRMANAGER_USE_SSL), 0)
-ADDITIONAL_LIBS = $(shell pkg-config --libs openssl)
+ADDITIONAL_LIBS = $(shell pkg-config.real --libs $(STAGING_DIR)/usr/lib/pkgconfig/openssl.pc)
 DEFINES += -DVDRMANAGER_USE_SSL=$(VDRMANAGER_USE_SSL)
 endif
 
 VDRMANAGER_USE_ZLIB := 1
 ifneq ($(VDRMANAGER_USE_ZLIB), 0)
-ADDITIONAL_LIBS += $(shell pkg-config --libs zlib)
+ADDITIONAL_LIBS += $(shell pkg-config.real --libs $(STAGING_DIR)/usr/lib/pkgconfig/zlib.pc)
 DEFINES += -DVDRMANAGER_USE_ZLIB=$(VDRMANAGER_USE_ZLIB)
 endif
 
