--- a/v4l/scripts/make_makefile.pl
+++ b/v4l/scripts/make_makefile.pl
@@ -143,69 +143,6 @@ sub open_makefile($) {
 	close $in;
 }
 
-my %obsolete;
-sub getobsolete()
-{
-	open OBSOLETE, '<obsolete.txt' or die "Can't open obsolete.txt: $!";
-	while (<OBSOLETE>) {
-		next if (/^\s*#/ || /^\s*$/);
-		chomp;
-		if (m|^(.*)/([^/]*)$|) {
-			$obsolete{$1}{"$2.ko"} = 1;
-		} else {
-			print "Unable to parse obsolete.txt:$.\n$_\n";
-		}
-	}
-
-	close OBSOLETE;
-}
-
-sub removeobsolete()
-{
-	while ( my ($dir, $files) = each(%obsolete) ) {
-		print OUT "\t\@if [ -d \$(DESTDIR)\$(KDIR26)/$dir ]; then echo -e \"\\nRemoving obsolete files from \$(DESTDIR)\$(KDIR26)/$dir:\"; fi\n";
-		print OUT "\t\@files='", join(' ', keys %$files), "'; ";
-
-		print OUT "for i in \$\$files;do if [ -f \"\$(DESTDIR)\$(KDIR26)/$dir/\$\$i\" ]; then ";
-		print OUT "echo -n \"\$\$i \";";
-		print OUT " rm \$(DESTDIR)\$(KDIR26)/$dir/\$\$i; fi; done; ";
-
-		print OUT "for i in \$\$files;do if [ -f \"\$(DESTDIR)\$(KDIR26)/$dir/\$\$i.(gz|bz2|xz)\" ]; then ";
-		print OUT "echo -n \"\$\$i.* \";";
-		print OUT " rm \$(DESTDIR)\$(KDIR26)/$dir/\$\$i.*; fi; done\n\n";
-		print OUT "\t\@if [ -d \$(DESTDIR)\$(KDIR26)/$dir ]; then echo; echo; fi\n";
-	}
-}
-
-#
-# Special hack for Ubuntu with their non-standard dirs
-#
-sub removeubuntu($)
-{
-	my $udir = shift;
-	my $dest = "/lib/modules/\$(KERNELRELEASE)/$udir";
-	my $filelist;
-
-	while ( my ($dir, $files) = each(%srcdir) ) {
-		$filelist .= ' '. join(' ', keys %$files);
-	}
-	while ( my ($dir, $files) = each(%obsolete) ) {
-		$filelist .= ' ' . join(' ', keys %$files);
-	}
-	$filelist =~ s/\s+$//;
-
-	print OUT "\t\@if [ -d $dest ]; then ";
-	print OUT "printf \"\\nHmm... distro kernel with a non-standard place for module backports detected.\\n";
-	print OUT "Please always prefer to use vanilla upstream kernel with V4L/DVB\\n";
-	print OUT "I'll try to remove old/obsolete LUM files from $dest:\\n\"; ";
-	print OUT "files='", $filelist, "'; ";
-
-	print OUT "for i in \$\$files;do find \"$dest\" \-name \"\$\$i\" \-exec echo \'{}\' \';\' ;";
-	print OUT " find \"$dest\" \-name \"\$\$i\" \-exec rm \'{}\' \';\' ;";
-	print OUT " done;";
-	print OUT " fi\n";
-}
-
 sub parse_dir()
 {
 	my $file = $File::Find::name;
@@ -216,76 +153,16 @@ sub parse_dir()
 
 ##############################################################################
 
-getobsolete();
-
 open OUT, '>Makefile.media' or die 'Unable to write Makefile.media';
 open_makefile('../linux/drivers/media/Makefile');
 
 find({wanted => \&parse_dir, no_chdir => 1}, '../linux/drivers/misc');
 
-# Creating Install rule
-print OUT "media-install:: media-rminstall\n";
-
-removeobsolete();
-removeubuntu("kernel/ubuntu/media");
-removeubuntu("kernel/ubuntu/lirc");
-removeubuntu("/updates/dkms");
-
-print OUT "\t\@echo \"Installing kernel modules under \$(DESTDIR)\$(KDIR26)/:\"\n";
-
-# change source dirs (relative to v4l dir)
-# into install dirs  (relative to DESTDIR/KDIR26)
-while (my ($dir, $files) = each %srcdir) {
-	my $idir = $dir;
-	$idir =~ s|\.\./linux/drivers/|../|;
-	$idir =~ s|\.\./media/?||;
-	$instdir{$idir} = $files;
-}
-
-while (my ($dir, $files) = each %instdir) {
-	print OUT "\t\@n=0;for i in ", join(' ', keys %$files), ";do ";
-	print OUT "if [ -f \"\$\$i\" ]; then ";
-	print OUT "if [ \$\$n -eq 0 ]; then ";
-	print OUT "echo -n \"\t$dir/: \"; ";
-	print OUT "install -d \$(DESTDIR)\$(KDIR26)/$dir; fi; ";
-	print OUT "n=\$\$\(\(\$\$n+1\)\); ";
-	print OUT "if [  \$\$n -eq 4 ]; then echo; echo -n \"\t\t\"; n=1; fi; ";
-	print OUT "echo -n \"\$\$i \"; ";
-	print OUT "install -m 644 -c \$\$i \$(DESTDIR)\$(KDIR26)/$dir; fi; done; ";
-	print OUT "if [  \$\$n -ne 0 ]; then echo; ";
-	print OUT "strip --strip-debug \$(DESTDIR)\$(KDIR26)/$dir/*.ko; ";
-	print OUT "fi;\n\n";
-}
-print OUT "\t@echo\n";
-print OUT "\t/sbin/depmod -a \$(KERNELRELEASE) \$(if \$(DESTDIR),-b \$(DESTDIR))\n\n";
-
-# Creating Remove rule
-print OUT "media-rminstall::\n";
-
-removeobsolete();
-removeubuntu("/ubuntu/media");
-removeubuntu("/updates/dkms");
-
-while ( my ($dir, $files) = each(%instdir) ) {
-	print OUT "\t\@if [ -d \$(DESTDIR)\$(KDIR26)/$dir ]; then echo -e \"\\nRemoving old \$(KDIR26)/$dir files:\"; fi\n";
-	print OUT "\t\@files='", join(' ', keys %$files), "'; ";
-
-	print OUT "for i in \$\$files;do if [ -f \"\$(DESTDIR)\$(KDIR26)/$dir/\$\$i\" ]; then ";
-	print OUT "echo -n \"\$\$i \";";
-	print OUT " rm \$(DESTDIR)\$(KDIR26)/$dir/\$\$i; fi; ";
-
-	print OUT "if [ -f \$(DESTDIR)\$(KDIR26)/$dir/\$\$i.* ]; then ";
-	print OUT "echo -n \"\$\$i.* \";";
-	print OUT " rm \$(DESTDIR)\$(KDIR26)/$dir/\$\$i.*; fi; done\n\n";
-	print OUT "\t\@if [ -d \$(DESTDIR)\$(KDIR26)/$dir ]; then echo; fi\n";
-}
-
 my $mediadeps = join(" \\\n", map("\t$_/Makefile", keys %srcdir ));
 $mediadeps =~ s,\.\./linux/drivers/media/\.\.,..,g;
 
 # Print dependencies of Makefile.media
 print OUT "Makefile.media: ../linux/drivers/media/Makefile \\\n";
-print OUT "\tobsolete.txt \\\n";
 print OUT $mediadeps;
 print OUT "\n";
 close OUT;
